VRDR/DeathRecord.cs

```diff
@@ -389,13 +389,15 @@ private void CreateDeathDateObs()
             DeathDateObs.Meta.Profile = deathdate_profile;
             DeathDateObs.Status = ObservationStatus.Final;
             DeathDateObs.Code = new CodeableConcept(CodeSystems.LOINC, "81956-5", "Date+time of death", null);
-            DeathDateObs.Subject = new ResourceReference("urn:uuid:" + Decedent.Id);
+            DeathDateObs.Subject = new ResourceReference("urn:uuid:" + (Decedent!=null?Decedent.Id:"000000000"));
             // A DeathDate can be represented either using the PartialDateTime or the valueDateTime; we always prefer
             // the PartialDateTime representation (though we'll correctly read records using valueDateTime) and so we
             // by default set up all the PartialDate extensions with a default state of "data absent"
             DeathDateObs.Value = new FhirDateTime();
             DeathDateObs.Value.Extension.Add(NewBlankPartialDateTimeExtension(true));
-            AddReferenceToComposition(DeathDateObs.Id, "DeathInvestigation");
+            if(Decedent!=null){
```

Comment:
Checking the Decedent to determine whether to add this reference is pretty opaque; once you make the change to not set the DeathDateObs.Subject if there isn't a Decedent just check to see if the DeathDateObs has a Subject set.